packages/advanced-logic/specs/payment-network-eth-proxy-contract-0.1.0-DRAFT.md

```diff
@@ -0,0 +1,219 @@
+# Payment Network - Ethereum - proxy contract
+
+You may be interested in this document if:
+
+- you want to create your own implementation of the Request protocol
+- you are curious enough to dive and see what is under the hood of the Request protocol
+
+Prerequisite: Having read the advanced logic specification (see [here](./advanced-logic-specs-0.1.0-DRAFT.md)).
+
+## Description
+
+This extension allows the payments and the refunds to be made in Ethereum tokens on the Ethereum blockchain.
+The payment is made through a proxy contract. This proxy contract does the Ethereum token transfer on behalf of
+the user. The contract ensures a link between an Ethereum transfer and a request through a `paymentReference`.
+This `paymentReference` consists of the last 8 bytes of a salted hash of the requestId: `last8Bytes(hash(lowercase(requestId + salt + address)))`:
+
+- `requestId` is the id of the request
+- `salt` is a random number with at least 8 bytes of randomness. It must be unique to each request
+- `address` is the payment address for payments, the refund address for refunds
+- `lowercase()` transforms all characters to lowercase
+- `hash()` is a keccak256 hash function
+- `last8Bytes()` take the last 8 bytes
+
+As a payment network, this extension allows to deduce a payment `balance` for the request. (see
+[Interpretation](#Interpretation))
+
+## Contract
+
+The contract contains one function called `transferWithReference` which takes 2 arguments:
+
+- `to` is the destination address
+- `paymentReference` is the reference data used to track the transfer (see `paymentReference`)
+
+The `TransferWithReference` event is emitted when the tokens are transfered. This event contains the same 2 arguments as the `transferWithReference` function plus the `amount` of ethereum sent.
+
+[See smart contract source](https://github.com/RequestNetwork/requestNetwork/blob/master/packages/smart-contracts/src/contracts/EthereumProxy.sol)
```

Comment:
404 for: https://github.com/RequestNetwork/requestNetwork/blob/master/packages/smart-contracts/src/contracts/EthereumProxy.sol

Reply:
Good point, The file is coming with the next PR.